lib/pbench/test/unit/agent/test_tool_meister.py

```diff
@@ -57,37 +82,37 @@ class TestCreateTar:
         "tools": [],
     }
 
-    def test_create_tar(self, agent_setup, mock_tar):
+    def test_create_tar(self, agent_setup, mock_tar, caplog):
         """Test create tar file"""
+        self.logger.setLevel(level=logging.ERROR)
         tm = ToolMeister(
             pbench_install_dir=None,
             tmp_dir=None,
             tar_path="tar_path",
             sysinfo_dump=None,
-            params=TestCreateTar.tm_params,
+            params=self.tm_params,
             redis_server=None,
             logger=None,
         )
         tmp_dir = agent_setup["tmp"]
-        tar_file = "test.tar.xz"
-
         cp = tm._create_tar(tmp_dir, pathlib.Path(tar_file))
         assert cp.returncode == 0
         assert cp.stdout == b""
 
-    def test_create_tar_ignore_warnings(self, agent_setup, mock_tar_no_warnings):
+    def test_create_tar_ignore_warnings(
+        self, agent_setup, mock_tar_no_warnings, caplog
```

Comment:
So does this mean that just bringing in the `caplog` fixture is enough to suppress pytest's display of the `WARNING` logs? Or are is this implicitly counting on inheriting the effect of the `self.logger.setLevel` from the previous test? (Which would be a bad idea.)

If you're sharing a logger across test cases, and using `setLevel`, be sure that you set it back. Or simply use `Logger.getLogger` in each test case where you need it, and just set that logger's level locally.

Reply:
Yeah the logger was really taking effect across the test functions, so I created a logger fixture that just yields the logger by temporarily setting the log level and then set it back to `NOTSET` inside the fixture.